src/Algebra/Graph/NonEmpty.hs

```diff
@@ -440,8 +495,14 @@ vertexList1 = NonEmpty.fromList . T.vertexList
 -- edgeList . 'edges1'       == 'Data.List.nub' . 'Data.List.sort' . 'Data.List.NonEmpty.toList'
 -- edgeList . 'transpose'    == 'Data.List.sort' . map 'Data.Tuple.swap' . edgeList
 -- @
+{-# RULES "edgeList/Int" edgeList = edgeIntList #-}
+{-# INLINE[1] edgeList #-}
 edgeList :: Ord a => NonEmptyGraph a -> [(a, a)]
-edgeList = T.edgeList
+edgeList = AM.edgeList . foldg1 AM.vertex AM.overlay AM.connect
+
+-- | Like 'edgeList' but specialised for NonEmptyGraph with vertices of type 'Int'.
+edgeIntList :: NonEmptyGraph Int -> [(Int,Int)]
+edgeIntList = AIM.edgeList . foldg1 AIM.vertex AIM.overlay AIM.connect
```

Comment:
Presumably, `foldg1 AM.vertex AM.overlay AM.connect` and `foldg1 AIM.vertex AIM.overlay AIM.connect` will appear in the `instance ToGraph`, so you could simply write `toAdjacencyIntMap` here?

Reply:
> Presumably, foldg1 AM.vertex AM.overlay AM.connect and foldg1 AIM.vertex AIM.overlay AIM.connect will appear in the instance ToGraph

You are right for `edgeList`, it is a mistake.

For `edgeIntList` there is only `adjacencyIntMap` in `ToGraph`, so I cannot use it for `AM.edgeSet` since `adjacencyIntMap` only return a `IntMap IntSet`.

I can add `toAdjacencyIntMap` in `ToGraph` if you want.

Reply:
Thanks, you are right, this seems to work only with `edgeList`.